ex5B/ex5.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Write a function that given a string, returns the first character that
+ * doesn't repeat in the whole string. Otherwise (all characters repeat),
+ * should return null.
+ *
+ * For example:
+ * firstUniqueChar("abcdedcba") === "e"
+ * firstUniqueChar("abcdeedcba") === null
+ * firstUniqueChar("abababcab") === "c"
+ * firstUniqueChar("aaaaaaaaa") === null
+ * firstUniqueChar("a") === "a"
+ */
+function firstUniqueChar(str) {
+
+  var objCharCounter = {}, objFirstChar = { charVal: null }, searchIndex = 0;
+
+  for (searchIndex = 0; searchIndex < str.length; searchIndex++) {
+
+    if (objCharCounter[str.charAt(searchIndex)] === undefined) {
+
+      // Create new property
+      objCharCounter[str.charAt(searchIndex)] = { index: searchIndex, count: 1};
+
+    } else {
+
+      // Add to property
+      objCharCounter[str.charAt(searchIndex)].count += 1;
+
+   }
+
+  }
+
+  charArray = Object.keys(objCharCounter);
+
+  for (searchIndex = 0; searchIndex < Object.keys(objCharCounter).length; searchIndex++) {
+
+		if (objCharCounter[charArray[searchIndex]].count === 1) {
+			if (objFirstChar.charVal === null || objCharCounter[charArray[searchIndex]].index < objFirstChar.charVal){
```

Comment:
`objFirstChar.charVal === null` is only used for initialization. In this case, why did you have it initialized as `objFirstChar = { charVal: null }`? Why not just `objFirstChar = null` and changing the condition here to `if (objFirstChar === null || ...`